polytope/polytope.py

```diff
@@ -201,19 +201,40 @@ def __copy__(self):
         P.fulldim = self.fulldim
         return P
 
+    def contains(self, point, abs_tol=ABS_TOL):
+        """Return True if polytope contains points.
+
+        @type point: array_like, 1d array, or 2d array (of column vectors)
+
+        @rtype: bool, 1d array
+
+        See Also
+        ========
+        L{is_inside}
+        """
+        if not isinstance(point, np.ndarray):
+            point = np.array(point)
+        if point.ndim == 1:
+            point.shape = (point.size, 1)
+        assert point.shape[0] == self.dim, "points should be col vectors"
+        test = self.A.dot(point) - self.b.reshape((self.b.size, 1)) < abs_tol
+        return np.all(test, axis=0)
+
     def __contains__(self, point, abs_tol=ABS_TOL):
         """Return True if polytope contains point.
 
         See Also
         ========
-        L{is_inside}
+        L{is_inside}, L{Polytope.contains}
         """
         if not isinstance(point, np.ndarray):
             point = np.array(point)
         test = self.A.dot(point.flatten()) - self.b < abs_tol
         return np.all(test)
 
     def are_inside(self, points, abs_tol=ABS_TOL):
+        warnings.warn('polytope.are_inside is deprecated.'
```

Comment:
"Return True if Region contains point." in the docstring of method `contains` can be improved in at least two ways:

1. By using "is inside" (or "is strictly inside", though "strictly" may already be suggested by "inside"), so that `contains` is not defined as "contains".
2. "point" originates from `__contains__`, which accepts a single point. The method `contains` accepts one or more points, so "point(s)" appears to be more descriptive.

`are_within` could be a way of expressing nonstrict inequality (`<=`), or perhaps `not_outside`. However, nonstrict inequality is irrelevant in a numerical context, so the user should expect `<` unless otherwise noted (there are situations were a polytope has boundaries at integer coordinates, and thus equality testing may make more sense, but those arrangements are of measure zero).

`contains` seems to be a good choice. Its similarity to `__contains__` suggests that the two methods have some meaning in common. In the other direction, a user may more easily guess that `contains` may be related to what `__contains__` does, and then consult the docstring of `contains`.

The method `__contains__` already doesn't implement containment in a container, so the user should expect `contains`, too, to mean something non-container-related.

`region.are_inside(points)` is unreadable. `region.contains(points)` is readable.

(Also: corrected remark to [`are_inside(point)` above](https://github.com/tulip-control/polytope/pull/45#discussion_r126788880).)